processor/groupbyattrsprocessor/processor.go

```diff
@@ -222,6 +222,29 @@ func getMetricInInstrumentationLibrary(ilm pmetric.ScopeMetrics, searchedMetric
 	metric.SetName(searchedMetric.Name())
 	metric.SetUnit(searchedMetric.Unit())
 
+	// Move other special type specific values
+	switch metric.DataType() {
+
+	case pmetric.MetricDataTypeHistogram:
+		metric.Histogram().SetAggregationTemporality(searchedMetric.Histogram().AggregationTemporality())
+
+	case pmetric.MetricDataTypeExponentialHistogram:
+		metric.ExponentialHistogram().SetAggregationTemporality(searchedMetric.ExponentialHistogram().AggregationTemporality())
+
+	case pmetric.MetricDataTypeSum:
+		metric.Sum().SetAggregationTemporality(searchedMetric.Sum().AggregationTemporality())
+
+	case pmetric.MetricDataTypeGauge:
+	case pmetric.MetricDataTypeSummary:
+
+	default:
+		gap.logger.Error("Unhandled metric data type.",
```

Comment:
Since this can be logged a lot if happens, I think it would be good to ensure we don't cause flood here. One solution is to [copy approach similar to this one](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L61-L78)

Reply:
To add further context, I think the components are using the logger defined here: https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/internal/telemetrylogs/logger.go#L31-L34 - so we're already somewhat covered

Reply:
That's still a lot and meant only as a general protection. Given that we know this can happen a lot, we should really just log it once per data dimension (metric name? some resource attribute?).

Reply:
The only way this happens is if there is a brand new metric type introduced into the ecosystem....and that metric type gets used by a target system with the groupbyattrs. I can move it, but I think a noisy error might be good, especially with the sampling already in place at a higher level.

Or I guess another way is if somehow a None metric type gets through here? but that's probably indicative of a bigger issue.

Reply:
This must be a unreachable part of the code. `getMetricInInstrumentationLibrary -> getGroupedMetricsFromAttributes` are called from cases related particular data type in `processMetrics`. So this seems to be not the right place to catch invalid metrics. Let's remove `default` along with other cases where there is no action

Reply:
Originally I had only the cases affected.  @pmm-sumo said it would be a good idea to capture this per: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9088/files/d2f93c94193f04a51d9245022a678c0b02192484#r842907003
This code should never be called unless the spec changes.  I'm ambivalent as to have this code or not, so will defer to you guys.

Reply:
Yeah, it's unreachable now but if other metric data type would be added in the future and this part would not be updated, we want to have a mechanism to notify about it

Reply:
Not only spec. Call of this function is leads to these switch cases: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/groupbyattrsprocessor/processor.go#L130-L164. So in order to run into this code another case statement is needed there. If we want to catch invalid metrics, I don't believe this is the right place. If we really want to do that, I think it's better to add it higher the call stack, e.g. in the switch case that I mentioned

Reply:
As discussed on the sampled logger, we don't want to handle that today. So from that perspective, we can move it to "debug" level and maybe leave a TODO item.

Having said that, I now see that if we really want do it right, then it should be in a different place, as @dmitryax pointed out.

Taking it all together, I'm leaning towards removing this section and maybe opening an issue on handling the unexpected values better (and addressing it in a separate PR)

Reply:
Sounds like we have an agreement here. @crobertson-conga can you please remove the `default` case along with other cases with no action

Reply:
Cool, for some reason I didn't see this comment a few days ago.  Removed the sampling code, this should be good to go.